sim/core/periodic_action.go

```diff
@@ -15,21 +15,12 @@ type DelayedActionOptions struct {
 }
 
 func NewDelayedAction(sim *Simulation, options DelayedActionOptions) *PendingAction {
-	pa := &PendingAction{
```

Comment:
it's simpler to panic() if `options.OnAction` is nil here...

Reply:
Done!